dali/test/tensor_adapter.h

```diff
@@ -0,0 +1,36 @@
+#ifndef DALI_TENSOR_BATCH_ADAPTER_H
+#define DALI_TENSOR_BATCH_ADAPTER_H
+
+#include "../pipeline/data/tensor_list.h"
+#include <vector>
+
+namespace dali {
+
+namespace testing {
+
+template<typename UnderlyingDataType>
+class TensorAdapter {
+  static_assert(std::is_fundamental<UnderlyingDataType>::value, "Underlying data type for tensor has to be fundamental");
+public:
+  TensorAdapter(std::vector<UnderlyingDataType> data, std::vector<size_t> shape) noexcept {}
```

Comment:
We should have a TensorListAdapter as a first-class entity. Here, we can't define an input of varying size - at least, not without much hassle.

Reply:
...or perhaps we shouldn't have this adapter at all...

Reply:
What do you mean by "input of varying size"? Is it "input with varying shape"?
If so, it is possible: please note that using `ToTensorList` you provide vector of `TensorAdapters`, where every one of them contains info about its shape. The `dali::TensorList` is then put together